microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -278,6 +278,21 @@ def _install_dependencies(self) -> None:
         cwd = node.working_path
 
         if isinstance(node.os, Ubuntu):
+            # DPDK requires backports channel for all releases
```

Comment:
1. I created a method in operating_system for debian. Can you improve it to support the ppa? You can merge the current parameters into one to get it free format.
2. In my test cases, it doesn't need the `-y`. Did you is it must have?
3. The `apt-get update` will be called after add repo. It doesn't need to be called again.

Reply:
I think we do need the 'y' since there's a splash screen that pops up telling you about what the backports repo is when you run this.

Reply:
I modified add_repository to special case "ppa:*" urls, this is nicer :) Thanks!

Reply:
How about we just leave one parameter like below? Some parameters are not used by ppa, if it's specified, that would be confused. You just need to update the xdp test case. It's the only one is using it so far.

```python
    def add_repository(
        self,
        repo: str,
        key_location: str = "",
```